community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/StoreMigrator.java

```diff
@@ -671,71 +669,36 @@ public void moveMigratedFiles( File migrationDir, File storeDir, String versionT
             //This means that we had no files only present in the page cache, this is fine.
         }
 
-        // update necessary neostore records
-        LogPosition logPosition = readLastTxLogPosition( migrationDir );
-        updateOrAddNeoStoreFieldsAsPartOfMigration( migrationDir, storeDir, versionToUpgradeTo, logPosition );
-
         // delete old logs
         legacyLogs.deleteUnusedLogFiles( storeDir );
     }
 
-    @Override
-    public void rebuildCounts( File storeDir, String versionToMigrateFrom, String versionToMigrateTo ) throws
-            IOException
-    {
-        if ( countStoreRebuildRequired( versionToMigrateFrom ) )
-        {
-            // create counters from scratch
-            Iterable<StoreFile> countsStoreFiles =
-                    Iterables.iterable( StoreFile.COUNTS_STORE_LEFT, StoreFile.COUNTS_STORE_RIGHT );
-            StoreFile.fileOperation( DELETE, fileSystem, storeDir, storeDir,
-                    countsStoreFiles, true, null, StoreFileType.STORE );
-            File neoStore = new File( storeDir, DEFAULT_NAME );
-            long lastTxId = MetaDataStore.getRecord( pageCache, neoStore, Position.LAST_TRANSACTION_ID );
-            rebuildCountsFromScratch( storeDir, lastTxId, versionToMigrateTo, pageCache );
-        }
-    }
-
-    boolean countStoreRebuildRequired( String versionToMigrateFrom )
-    {
-        return StandardV2_3.STORE_VERSION.equals( versionToMigrateFrom ) ||
-                StandardV3_0.STORE_VERSION.equals( versionToMigrateFrom ) ||
-                StoreVersion.HIGH_LIMIT_V3_0_0.versionString().equals( versionToMigrateFrom ) ||
-                StoreVersion.HIGH_LIMIT_V3_0_6.versionString().equals( versionToMigrateFrom ) ||
-                StoreVersion.HIGH_LIMIT_V3_1_0.versionString().equals( versionToMigrateFrom );
-    }
-
-    private void rebuildCountsFromScratch( File storeDir, long lastTxId, String versionToMigrateTo,
-                                           PageCache pageCache )
+    private void updateOrAddNeoStoreFieldsAsPartOfMigration( File migrationDir, File storeDir,
+            String versionToMigrateTo, LogPosition lastClosedTxLogPosition ) throws IOException
     {
-        final File storeFileBase = new File( storeDir, MetaDataStore.DEFAULT_NAME + StoreFactory.COUNTS_STORE );
-
-        RecordFormats recordFormats = selectForVersion( versionToMigrateTo );
-        StoreFactory storeFactory = new StoreFactory( storeDir, pageCache, fileSystem, recordFormats,
-                NullLogProvider.getInstance() );
-        try ( NeoStores neoStores = storeFactory.openAllNeoStores() )
+        final File storeDirNeoStore = new File( storeDir, DEFAULT_NAME );
+        final File migrationDirNeoStore = new File( migrationDir, DEFAULT_NAME );
+        // We use the page cache for copying the neostore since it might be on a block device.
+        int pageSize = pageCache.pageSize();
+        try ( PagedFile fromFile = pageCache.map( storeDirNeoStore, pageSize );
```

Comment:
If I get this right...
Instead of updating 'neostore' straight into the store directory that we are migrating from (after migration is done and we have copied everything back here) we now copy `neostore` to migration directory and do update there before copying migrated store back to store directory. Correct?
If so, nice. Seems weird to continue migration after the we have copied everything back.

Could we extract this as something like `copyFile`?